packages/mobile/src/pincode/PincodeSet.tsx

```diff
@@ -109,9 +136,23 @@ export class PincodeSet extends React.Component<Props, State> {
     if (this.isPin1Valid(pin1) && this.isPin2Valid(pin2)) {
       setCachedPin(DEFAULT_CACHE_ACCOUNT, pin1)
```

Comment:
For the `changePin` case, it seems we should only call this when `updatePin` is successful, right?

Reply:
Yeah, good catch, fixed!